include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -1021,9 +1023,9 @@ __pattern_search_n(__parallel_tag<_IsVector> __tag, _ExecutionPolicy&& __exec, _
 {
     if (static_cast<_Size>(__last - __first) == __count)
     {
-        const bool __result =
-            !__internal::__pattern_any_of(__tag, ::std::forward<_ExecutionPolicy>(__exec), __first, __last,
-                                          [&__value, &__pred](const _Tp& __val) { return !__pred(__val, __value); });
+        const bool __result = !__internal::__pattern_any_of(__tag, ::std::forward<_ExecutionPolicy>(__exec), __first,
+            __last, [&__value, __pred](auto&& __val) mutable { return !__pred(std::forward<decltype(__val)>(__val),
```

Comment:
What was the purpose of switching from `const auto& __val` to `auto&& __val`? I suppose `const auto& __val` is better, because the underlying data should not be changed by any search algorithm.

Reply:
I don't remember for sure... probably, a reason also was `NonConstAdapter`.

Reply:
I've successfully built the range tests using icpx/libstdc++13 and rolling back only this change (on this branch and on MVP2 branch).

The change looks unnecessary, but is not harmful at the same time. It is done, so I do not mind leaving it here.